app/src/main/kotlin/com/wire/android/di/GetIncomingCallsUseCaseProvider.kt

```diff
@@ -0,0 +1,22 @@
+package com.wire.android.di
+
+import com.wire.kalium.logic.CoreLogic
+import com.wire.kalium.logic.data.user.UserId
+import com.wire.kalium.logic.feature.call.usecase.GetIncomingCallsUseCase
+import com.wire.kalium.logic.feature.message.GetNotificationsUseCase
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedFactory
+import dagger.assisted.AssistedInject
+
+class GetIncomingCallsUseCaseProvider @AssistedInject constructor(
```

Comment:
Maybe we could just create such provider for the whole session scope? Because we're getting more and more assisted providers for use cases and all of them require session scope, so we could replace all of them (`GetIncomingCallsUseCaseProvider`, `ListenToEventsUseCaseProvider`, `GetNotificationsUseCaseProvider`) with just single `SessionScopeProvider` - what do you think? 
It also would be the similar approach we already have with `ClientScope` - we have single assisted `ClientScopeProvider` and multiple usecase providers in `CoreLogicModule` that take advantage of this assisted scope provider

Reply:
yeah, i was thinking about something like that :)